modules/installation-azure-user-defined-routing.adoc

```diff
@@ -0,0 +1,84 @@
+program is not responsible for configuring this.
+
+When configuring a cluster to use user-defined routing, the installation program
+does not create the following resources:
+
+* Outbound rules for access to Internet.
+* Public IPs for the public load balancer.
+* Kubernetes Service object to add the cluster machines to the public load
+balancer for outbound requests.
+
+You must ensure the following items are available before setting user-defined
+routing:
+
+* Egress to the internet is possible to pull container images, unless using an
+internal registry mirror.
+* The cluster can access Azure APIs.
+* Various allowlist endpoints are configured. You can reference these endpoints
+in the *Configuring your firewall* section.
+
+There are several pre-existing networking setups that are supported for internet
+access using user-defined routing.
+
+[discrete]
+== Private cluster with network address translation
+
+You can use link:https://docs.microsoft.com/en-us/azure/virtual-network/nat-overview[Azure VNET network address translation (NAT)]
+to provide outbound internet access for the subnets in your cluster. You can
+reference
+link:https://docs.microsoft.com/en-us/azure/virtual-network/quickstart-create-nat-gateway-cli[Create a NAT gateway using Azure CLI]
+in the Azure documentation for configuration instructions.
+
+When using a VNet setup with Azure NAT and user-defined routing configured, you
+can create a private cluster with no public endpoints. 
+
+[discrete]
+== Private cluster with Azure Firewall
+
+You can use Azure Firewall to provide outbound routing for the VNet used to
+install the cluster. You can learn more about
+link:https://docs.microsoft.com/en-us/azure/aks/egress-outboundtype#deploy-a-cluster-with-outbound-type-of-udr-and-azure-firewall[providing user-defined routing with Azure Firewall]
+in the Azure documentation.
+
+When using a VNet setup with Azure Firewall and user-defined routing configured,
+you can create a private cluster with no public endpoints.
+
+[discrete]
+== Private cluster with a proxy configuration
+
+You can use a proxy with user-defined routing to allow egress to the internet.
+You must make sure that cluster Operators do not access Azure APIs using a
+proxy; Operators must have access to Azure APIs outside of the proxy.
+
+When using the default route table for subnets, with `0.0.0.0/0` populated
+automatically by Azure, all Azure API requests are routed over Azure's internal
+network even though the IP addresses are public. As long as the Network Security
+Group rules allow egress to Azure API endpoints, proxies with user-defined
+routing configured allow you to create private clusters with no public
+endpoints.
+
+[discrete]
+== Private cluster with no internet access
+
+You can have VNets with no access to the internet if your cluster has access to
+the following:
+
+* An internal registry mirror that allows for pulling container images.
```

Comment:
I know consistency is probably the most important thing, but I thought you shouldn't put periods on bulleted lists if they weren't full sentences

Reply:
Good call; I'll remove them.